src/components/AddTodo/AddTodo.tsx

```diff
@@ -1,52 +1,46 @@
-import React, { useState } from 'react';
-import { TextField, Button, Box } from '@mui/material';
-import { makeStyles, createStyles } from '@mui/styles';
-import { useDispatch } from 'react-redux';
-import { addTodo } from '../../redux/todosSlice';
-
-const useStyles = makeStyles(() =>
-  createStyles({
-    addTodo: {
-      margin: '0 auto',
-      display: 'flex',
-      alignItems: 'center',
-      gap: 1,
-      padding: '10px',
-    },
-  })
-);
+import React, {useState} from 'react';
+import {TextField, Button, Box} from '@mui/material';
+import {useDispatch, useSelector} from 'react-redux';
+import {addTodo} from '../../redux/todosSlice';
+import {RootState} from '../../redux/store';
 
 const AddTodo = () => {
   const [todoText, setTodoText] = useState('');
+  const [addBtnDisabled, setAddBtnDisabled] = useState(true);
   const dispatch = useDispatch();
-  const classes = useStyles();
+  const todos = useSelector((state: RootState) => state.todosList.todos);
 
   const handleSubmit = (event: React.FormEvent<HTMLFormElement>) => {
     event.preventDefault();
     if (todoText.trim()) {
       dispatch(addTodo(todoText));
       setTodoText('');
+      setAddBtnDisabled(true);
     }
   };
 
   const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
-    setTodoText(event.target.value);
+    const value = event.target.value;
+    const addBtnIsDisabled = !value.length || todos.some(item => item.text === value);
+    setTodoText(value);
+    setAddBtnDisabled(addBtnIsDisabled);
   };
 
   return (
     <Box
       component='form'
       onSubmit={handleSubmit}
-      className={classes.addTodo}
+      display='flex' alignItems='center' gap={1}
```

Comment:
Pre-commit formatting doesn't work?

Reply:
It's odd because the husky pre-commit hook has been configured and in the PyCharm 'Run Git hooks' checkbox set to true.

![image](https://github.com/yauheni-ulyanovich/ToDoList/assets/18418716/aa995891-d570-4f7a-ba0b-36ec44318190)

Reply:
I added .husky/pre-commit file and now the hook works as expected